examples/project-delete/main.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+
+	projectClient "github.com/openshift/client-go/project/clientset/versioned"
+	"github.com/pkg/errors"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func getOpenShiftProjectClient(kubeconfig string) (*projectClient.Clientset, error) {
+	// kubeconfig passed as parameter is coming from cmd line
+
+	// if kubeconfig not provided on cmd line then read it from the env
+	if kubeconfig == "" {
+		kubeconfig = os.Getenv("KUBECONFIG")
+	}
+
+	// if it is still empty then take the value from the file which acts as default
+	if kubeconfig == "" {
+		kubeconfig = os.ExpandEnv("$HOME/.kube/config")
+	}
+
+	var (
+		err    error
+		config *rest.Config
+	)
+
+	// if kubeconfig is empty then in cluster config generation function will called
+	config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+	return projectClient.NewForConfig(config)
+}
+
+func errorsToError(errs []error) error {
+	var errStr string
+	for _, err := range errs {
+		errStr += fmt.Sprintf("%s\n", err)
+	}
+	return fmt.Errorf("%s", errStr)
+}
+
+func deleteUserProjects(cli *projectClient.Clientset, username string) error {
+	projects, err := cli.ProjectV1().Projects().List(meta_v1.ListOptions{})
+	if err != nil {
+		return errors.Wrap(err, "could not list projects")
```

Comment:
This doesn't work well in golang because the information about the original error is lost unless you know about this non-standard error to unwrap.  You return the bare error and explain in a comment it allows the k8s.io/apimachinery/pkg/api/errors.IsNotFound and the like to operate correctly.